Add support for the Avnet MaaXBoard (util_libs)

mjenkcapgemini opened the pull request:
This pull request is a call for feedback and comment on an initial implementation for supporting seL4 on the Avnet MaaXBoard, with the intention of the changes below being merged back to the main util_libs repository. This pull request is dependent on another pull request which provides the core support for the MaaXBoard in the seL4 repository; this can be found [here](https://github.com/seL4/seL4/pull/829).

This pull request contains the following changes:
* Updates to configuration files to include `KernelPlatformMaaxBoard` and `CONFIG_PLAT_MAAXBOARD` to support building seL4 for the MaaXBoard.

seL4test has been tested with these changes only on the MaaXBoard itself, but all tests run as expected resulting in a `All is well in the universe` message. No other platforms have been tested.

Test with: seL4/seL4#829, seL4/seL4_tools#144

axel-h commented on libplatsupport/plat_include/imx8m/platsupport/plat/serial.h:
```diff
@@ -24,7 +25,7 @@
 #define DEFAULT_SERIAL_PADDR        UART2_PADDR
 #define DEFAULT_SERIAL_INTERRUPT    UART2_IRQ
-#elif defined(CONFIG_PLAT_IMX8MQ_EVK)
+#elif defined(CONFIG_PLAT_IMX8MQ)
```
```suggestion
#elif defined(CONFIG_PLAT_IMX8MQ_EVK) || defined(CONFIG_PLAT_MAAXBOARD)
```
This cannot be generic, because there is no generic board. There might be common guidelines, but in the end it's always up to each specific boards that use a SOC, how things are set up there.

axel-h commented on libplatsupport/mach_include/imx/platsupport/mach/serial.h:
```diff
@@ -27,7 +28,7 @@ enum chardev_id {
 
 #elif defined(CONFIG_PLAT_WANDQ) || defined(CONFIG_PLAT_NITROGEN6SX) \
-      || defined(CONFIG_PLAT_IMX7_SABRE) || defined(CONFIG_PLAT_IMX8MQ_EVK)
+      || defined(CONFIG_PLAT_IMX7_SABRE) || defined(CONFIG_PLAT_IMX8MQ)
```
```suggestion
      || defined(CONFIG_PLAT_IMX7_SABRE) || defined(CONFIG_PLAT_IMX8MQ_EVK) \
      || defined(CONFIG_PLAT_MAAXBOARD)
```
Same here, this is really board specific and not a SOC or SOC-Family setting.

axel-h commented:
Can you squash the commits? One for the formatting changes and then one for the new board on top?

mjenkcapgemini commented:
> Can you squash the commits? One for the formatting changes and then one for the new board on top?

I have squashed the last two board-related commits. Please let me know if you meant me to do more.

axel-h commented:
> > Can you squash the commits? One for the formatting changes and then one for the new board on top?
> 
> I have squashed the last two board-related commits. Please let me know if you meant me to do more.

Basically the change in this PR look ok to me. Thus I meant sqaush all commits in this PR, so there are just two left: formatting, add board. This is the change set what we would merge (using rebase and fast-forward) then eventually.

mjenkcapgemini commented:
> > > Can you squash the commits? One for the formatting changes and then one for the new board on top?
> > 
> > 
> > I have squashed the last two board-related commits. Please let me know if you meant me to do more.
> 
> Basically the change in this PR look ok to me. Thus I meant sqaush all commits in this PR, so there are just two left: formatting, add board. This is the change set what we would merge (using rebase and fast-forward) then eventually.

OK thanks for clarifying. I have now squashed down to two commits.
